app/services/piece-access-level-service.js

```diff
@@ -131,6 +131,29 @@ export default class PieceAccessLevelService extends Service {
     }));
   }
 
+  async updateSubmissionAccessLevelOfSubcase(subcase) {
+    const vertrouwelijk = await this.store.findRecordByUri('concept', CONSTANTS.ACCESS_LEVELS.VERTROUWELIJK);
+    const submissionActivities = await this.store.query('submission-activity', {
+      'filter[subcase][:id:]': subcase.id,
+      include:'pieces',
+    });
+    const pieces = [];
+    for (const submissionActivity of submissionActivities.toArray()) {
+      let submissionPieces = await submissionActivity.pieces;
+      submissionPieces = submissionPieces.toArray();
+      pieces.push(...submissionPieces);
+    }
```

Comment:
This list of pieces from submissions will contain previous versions of each other. (and duplicates when subcase has been postponed)
Later on we update previous access levels so some pieces will be processed twice (second time no save).

We might be better off making a set with only the last versions of each and run that through a "strengthen" method (see other comment)